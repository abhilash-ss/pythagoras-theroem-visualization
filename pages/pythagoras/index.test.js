import React from "react";
import { shallow } from "enzyme";
import Pythagoras from "./index";

jest.mock("next/router", () => ({
  withRouter: (component) => component
}));

describe("Pythagoras", () => {
  let props;
  beforeEach(() => {
    props = {
      router: {
        push: jest.fn(),
        query: { a: 3, b: 4, c: 5 }
      }
    };
  });
  it("should render without crash", () => {
    const component = shallow(<Pythagoras {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should render with params", () => {
    props.router.query = { a: 4, b: 3, c: 5 };
    const component = shallow(<Pythagoras {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should render without params", () => {
    props.router.query = {};
    const component = shallow(<Pythagoras {...props} />);
    expect(component).toMatchSnapshot();
  });
});
