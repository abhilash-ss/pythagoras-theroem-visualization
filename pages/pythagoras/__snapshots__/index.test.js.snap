// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pythagoras should render with params 1`] = `ShallowWrapper {}`;

exports[`Pythagoras should render without crash 1`] = `ShallowWrapper {}`;

exports[`Pythagoras should render without params 1`] = `ShallowWrapper {}`;
